repose-aggregator/components/filters/attribute-mapping-policy-validation-filter/src/test/scala/org/openrepose/filters/attributemapping/AttributeMappingPolicyValidationFilterTest.scala

```diff
@@ -0,0 +1,241 @@
+      request.setMethod("PUT")
+
+      filter.validateHttpMethod(request) shouldBe a[Success[_]]
+    }
+
+    HttpMethods foreach { method =>
+      it(s"should return a Failure if the method is $method") {
+        request.setMethod(method)
+
+        filter.validateHttpMethod(request) shouldBe a[Failure[_]]
+      }
+    }
+  }
+
+  describe("getPolicyAsXmlSource") {
+    it(s"should return a Source if the content-type is JSON") {
+      filter.getPolicyAsXmlSource(MediaType.APPLICATION_JSON_VALUE, new ByteArrayInputStream(ValidJsonPolicy.getBytes)) shouldBe a[Success[_]]
+    }
+
+    it(s"should return a Source if the content-type is XML") {
+      filter.getPolicyAsXmlSource(MediaType.APPLICATION_XML_VALUE, new ByteArrayInputStream(ValidXmlPolicy.getBytes)) shouldBe a[Success[_]]
+    }
+
+    it("should return a Failure if the content-type is not supported") {
+      filter.getPolicyAsXmlSource("text/plain", new ByteArrayInputStream(ValidJsonPolicy.getBytes)) shouldBe a[Failure[_]]
+    }
+  }
+
+  describe("doFilter") {
+    it("should return a 415 if the Content-Type is not supported") {
+      request.setMethod("PUT")
+      request.setContentType("text/plain")
+
+      filter.doFilter(request, response, chain)
+
+      response.isCommitted shouldBe true
+      response.getStatus shouldBe SC_UNSUPPORTED_MEDIA_TYPE
+    }
+
+    it("should forward the request if JSON validation succeeds") {
+      request.setMethod("PUT")
+      request.setContent(ValidJsonPolicy.getBytes)
+      request.setContentType(MediaType.APPLICATION_JSON_VALUE)
+
+      filter.doFilter(request, response, chain)
+
+      chain.getRequest should not be theSameInstanceAs(request)
+      chain.getResponse shouldBe theSameInstanceAs(response)
+    }
+
+    it("should forward the request if XML validation succeeds") {
+      request.setMethod("PUT")
+      request.setContent(ValidXmlPolicy.getBytes)
+      request.setContentType(MediaType.APPLICATION_XML_VALUE)
+
+      filter.doFilter(request, response, chain)
+
+      chain.getRequest should not be theSameInstanceAs(request)
+      chain.getResponse shouldBe theSameInstanceAs(response)
+    }
+
+    it("should return a 400 if the policy fails to validate") {
+      request.setMethod("PUT")
+      request.setContent(InvalidJsonPolicy.getBytes)
+      request.setContentType(MediaType.APPLICATION_JSON_VALUE)
+
+      filter.doFilter(request, response, chain)
+
+      response.getStatus shouldBe HttpServletResponse.SC_BAD_REQUEST
+      response.isCommitted shouldBe true
+    }
+
+    HttpMethods foreach { method =>
+      it(s"should pass $method requests through the filter") {
+        request.setMethod(method)
+
+        filter.doFilter(request, response, chain)
+
+        chain.getRequest shouldBe theSameInstanceAs(request)
+        chain.getResponse shouldBe theSameInstanceAs(response)
+      }
+    }
+  }
+}
+
+object AttributeMappingPolicyValidationFilterTest {
+  final val HttpMethods = Set(
```

Comment:
Oops. It used to be all HTTP methods, but I changed the tests a bit and that is what this became. I'll change it.